Jenkinsfile.integration

```diff
@@ -41,6 +27,29 @@ pipeline {
         USER = "jenkins" /* Why isn't this set in the jenkins environment? */
     }
 
+    parameters {
+        /* first value in the list is the default */
+        choice(choices: ['airship', 'osh'], description: 'Which deployment mechanism?', name: 'deployment')
+        choice(choices: ['caasp3', 'caasp4'], description: 'Which caasp version to use?', name: 'caasp_version')
+        booleanParam(name: 'reserve_env', description: 'Should we reserve the cloud_env resource?', defaultValue: true)
+        string(name: 'cloud_env', description:'The environment identifier', defaultValue: env.SOCOK8S_ENVNAME)
```

Comment:
We will probably want to change the `cloud_env` default value to the label we apply to the new lockable resources for the slots, but I'm confused about how this interacts with the `agent` directive.